test/multi_tenancy.sh

```diff
@@ -74,51 +80,60 @@ function create_user_and_assign_to_projects() {
+    local contenttype="-H Content-Type:application/json"
+
+    local payload="{\"log\":\"log message 1\",\"stream\":\"stderr\",\"time\":\"2014-09-25T21:15:03.499185026Z\",\"kubernetes\":{\"namespace_name\":\"$namespace\",\"pod_name\":\"synthetic-logger-0.25lps-pod\"},\"docker\":{\"container_id\":\"container123\"}}"
+    local alias="{\"actions\":[{\"add\":{\"index\":\"$index\",\"alias\":\"app\"}}]}"
+
+    # add index
+    os::log::debug $( curl_es_pod "$pod" "/$index/multi-tenancy-test/" -XPOST $xff $xocpns $contenttype -d "$payload" | python -mjson.tool 2>&1 )
+    # add alias
+    os::log::debug $( curl_es_pod "$pod" "/_aliases" -XPOST $xff $xocpns $contenttype -d "$alias" | python -mjson.tool 2>&1 )
+    
+    os::log::debug $(curl_es_pod "$pod" "/app/multi-tenancy-test/_search" -XGET | python -mjson.tool)
 }
 
 function test_user_has_proper_access() {
     local user=$1; shift
     local pw=$1; shift
-    local indices="["
-    local comma=""
-    # rest - indices to which access should be granted
-    for proj in "$@" ; do
-        os::log::info See if user $user can read /project.$proj.*
-        get_test_user_token $user $pw false
-        nrecs=$( curl_es_pod_with_token $espod "/project.$proj.*/_count" $test_token | \
-                     get_count_from_json )
-        if ! os::cmd::expect_success "test $nrecs = 1" ; then
-            os::log::error $user cannot access project.$proj.* indices
-            curl_es_pod_with_token $espod "/project.$proj.*/_count" $test_token | python -mjson.tool
-            os::log::info "acl documents"
-            oc exec -c elasticsearch $espod -- es_acl get --doc=roles
-            oc exec -c elasticsearch $espod -- es_acl get --doc=rolesmapping
-            exit 1
-        fi
-        indices="${indices}${comma}"'"'"project.$proj.*"'"'
-        comma=,
-    done
-    indices="${indices}]"
+    local alias=$1; shift
 
-    # test user has access for msearch for multiple indices
-    os::log::info See if user $user can _msearch "$indices"
+    os::log::info See if user $user can read alias /$alias
     get_test_user_token $user $pw false
-    nrecs=$( { echo '{"index":'"${indices}"'}'; echo '{"size":0,"query":{"match_all":{}}}'; } | \
-                     curl_es_pod_with_token_and_input $espod "/_msearch" $test_token -XPOST --data-binary @- | \
+    local xfuser='-H x-forwarded-user:'"$user"
+    local xocpns='-H x-ocp-namespace:'"$user_project_list"
+    local xfroles='-H x-forwarded-roles:project_user'
+    
+    nrecs=$( curl_es_pod_with_token $espod "/$alias/_count" $test_token -XGET $xfuser $xocpns $xfroles | get_count_from_json )
+    expected=${1:-$user_project_num}
+    if ! os::cmd::expect_success "test $nrecs = $expected" ; then
+        os::log::error $user cannot access alias /$alias
+        curl_es_pod_with_token $espod "/$alias/_count" $test_token -XGET $xfuser $xocpns $xfroles | python -mjson.tool
+        oc exec -c elasticsearch $espod -- es_acl get --doc=roles
```

Comment:
Consider writing these to a file in the artifact dir

Reply:
done